transport/src/test/java/io/netty/channel/embedded/EmbeddedChannelTest.java

```diff
@@ -316,6 +316,52 @@ public void testReleaseOutbound() {
         }
     }
 
+    @Test
+    public void testWriteLater() {
+        EmbeddedChannel channel = new EmbeddedChannel(new ChannelOutboundHandlerAdapter() {
+            @Override
+            public void write(final ChannelHandlerContext ctx, final Object msg, final ChannelPromise promise)
+                    throws Exception {
+                ctx.executor().execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        ctx.write(msg, promise);
+                    }
+                });
+            }
+        });
+        Object msg = new Object();
+
+        assertTrue(channel.writeOutbound(msg));
+        assertTrue(channel.finish());
+        assertSame(msg, channel.readOutbound());
+        assertNull(channel.readOutbound());
+    }
+
+    @Test
+    public void testWriteScheduled() throws InterruptedException  {
+        final int delay = 500;
+        EmbeddedChannel channel = new EmbeddedChannel(new ChannelOutboundHandlerAdapter() {
+            @Override
+            public void write(final ChannelHandlerContext ctx, final Object msg, final ChannelPromise promise)
+                    throws Exception {
+                ctx.executor().schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        ctx.writeAndFlush(msg, promise);
+                    }
+                }, delay, TimeUnit.MILLISECONDS);
+            }
+        });
+        Object msg = new Object();
+
+        assertFalse(channel.writeOutbound(msg));
+        Thread.sleep(delay  * 2);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2925)